BuildScript/BuildTasks/CreateTestConfigurations.cs

```diff
@@ -0,0 +1,154 @@
+    private TestConfigurationsFilter _testConfigurationFilter;
+
+    [Required]
+    public ITaskItem[] Input { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedBrowsers { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledBrowsers { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedDatabaseSystems { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledDatabaseSystems { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedPlatforms { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledPlatforms { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedExecutionRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledExecutionRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedTargetRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledTargetRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledConfigurationIDs { get; set; }
+
+    [Output]
+    public ITaskItem[] Output { get; set; }
+
+    public override bool Execute ()
+    {
+      if (_testConfigurationFactory == null)
+      {
+        _testConfigurationFactory = new TestConfigurationsFactory (
+            GetItemSpecCollection (SupportedBrowsers),
+            GetItemSpecCollection (SupportedDatabaseSystems),
+            GetItemSpecCollection (SupportedPlatforms),
+            new[] { "debug", "release" },
+            GetItemSpecCollection (SupportedExecutionRuntimes),
+            GetItemSpecCollection (SupportedTargetRuntimes));
+      }
+
+      if (_testConfigurationFilter == null)
+      {
+        _testConfigurationFilter = new TestConfigurationsFilter (
+            GetItemSpecCollection (SupportedBrowsers),
+            GetItemSpecCollection (EnabledBrowsers),
+            GetItemSpecCollection (SupportedDatabaseSystems),
+            GetItemSpecCollection (EnabledDatabaseSystems),
+            GetItemSpecCollection (SupportedPlatforms),
+            GetItemSpecCollection (EnabledPlatforms),
+            GetItemSpecCollection (SupportedExecutionRuntimes),
+            GetItemSpecCollection (EnabledExecutionRuntimes),
+            GetItemSpecCollection (SupportedTargetRuntimes),
+            GetItemSpecCollection (EnabledTargetRuntimes),
+            new[] { "debug", "release" },
+            GetItemSpecCollection (EnabledConfigurationIDs));
+      }
+
+      var output = new List<ITaskItem>();
+      foreach (var inputItem in Input)
+      {
+        var testAssemblyFullPath = inputItem.ItemSpec;
+        var semicolonSeparatedTestConfigurations = inputItem.GetMetadata ("TestConfiguration");
+        var testSetupFile = inputItem.GetMetadata ("TestSetupBuildFile");
+
+        try
+        {
+          var unfilteredTestConfigurations = _testConfigurationFactory.CreateTestConfigurations (
+              testAssemblyFullPath,
+              semicolonSeparatedTestConfigurations,
```

Comment:
I've moved the semicolon parsing up to the task, but I'm still not sure about the naming. I know that they can't all be "testConfigurations", and since they always come in plurals it's hard to name collections of them (testConfigurationsCollection?), especially since they're coming as strings from msbuild and technically aren't "collections". Open for suggestions @iam-mholle @MichaelKetting